cmd/integrationArtifactDeploy.go

```diff
@@ -159,12 +163,12 @@ func getHTTPErrorMessage(httpErr error, response *http.Response, httpMethod, sta
 }
 
 //getIntegrationArtifactDeployStatus - Get integration artifact Deploy Status
-func getIntegrationArtifactDeployStatus(config *integrationArtifactDeployOptions, httpClient piperhttp.Sender, apiHost string) (string, error) {
+func getIntegrationArtifactDeployStatus(config *integrationArtifactDeployOptions, httpClient piperhttp.Sender, apiHost string, taskId string) (string, error) {
 	httpMethod := "GET"
 	header := make(http.Header)
 	header.Add("content-type", "application/json")
 	header.Add("Accept", "application/json")
-	deployStatusURL := fmt.Sprintf("%s/api/v1/IntegrationRuntimeArtifacts('%s')", apiHost, config.IntegrationFlowID)
+	deployStatusURL := fmt.Sprintf("%s/api/v1/BuildAndDeployStatus(TaskId='%s')", apiHost, taskId)
```

Comment:
Just to confirm- the config.IntegrationFlowId is no longer needed for this version of the API? In the other parts of the code you just add taskId to the parameters but here you're replacing another one

Reply:
yes its not needed for new polling API arguments. but its still used for deploy API

Reply:
I don't really understand that.. is this not the deploy API?

Reply:
There are 3 APIs and this argument is not needed for this API